enhancements/ingress/ingress-operator-operand-logging-level.md

```diff
@@ -80,17 +74,50 @@ The Ingress Operator currently uses [Zapr](https://github.com/go-logr/zapr) logg
 are displayed in the Ingress Operator pod logs. Debug-level logs are currently omitted in the Ingress Operator pod logs, although, the Ingress Operator itself rarely logs at any level
 other than `log.Info` and `log.Error`. In general, more lower-level logging calls could be added to the Ingress Operator.
 
-For this enhancement, a new `OperatorLogLevel` field is added to the Ingress Config resource:
+In a previous iteration of this enhancement proposal, the `OperatorLogLevel` field was proposed to be included in the
+Ingress Config resource (ingresses.config.openshift.io). However, the intent of the Ingress Config resource is to define
+cluster-wide ingress policies and configuration, not to define specific configuration for the Ingress Operator. As such,
+a new Custom Resource Definition (CRD) must be defined for Ingress Operator configuration, called `Ingress`
```

Comment:
The proposal made me think of how the same can be achieved for an OLM operator. The OLM operator has a [subscription object](https://pkg.go.dev/github.com/operator-framework/api@v0.17.3/pkg/operators/v1alpha1#Subscription) which allows the user [to set environment variables](https://pkg.go.dev/github.com/operator-framework/api@v0.17.3/pkg/operators/v1alpha1#SubscriptionConfig). An environment variable for the log level can be defined in the operator's deployment ([example](https://github.com/openshift/node-observability-operator/blob/40aaadae753f6bf385c9a171933c75faa8723c8b/config/manager/manager.yaml#L38-L39)) and can be overridden in the Subscription by the user (`subscription.spec.config.env`). Then this environment variable can be [passed](https://github.com/openshift/node-observability-operator/blob/main/config/manager/manager.yaml#L34) to `--zap-log-level flag` (standard flag which [can be added by controller-runtime](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/log/zap#Options.BindFlags)).

I thought that the similar logic may be useful for any cluster operator too. Each cluster operator has its own `ClusterOperator` object whose `Spec` can be enhanced with something similar, say `.spec.config.env`. That may be a little more work as apart from the API change , [the cluster version operator](https://github.com/openshift/cluster-version-operator) would need to be modified. On the cluster ingress operator's side though the logic described above can be applied (+ [example of the logger flags binding](https://github.com/openshift/node-observability-operator/blob/main/main.go#L52-L60)).